### PR TITLE
MOD-5945: Fix bad comparison

### PR DESCRIPTION
**Pull request by raz-mon**
Fix of a bad condition. Today we don't fail due to a check of the alias list in case the `INDEXSPEC_LOAD_NOALIAS` flag isn't on, but there is no good reason comparing the length of the strings (index and alias) instead of the strings, which is prone for error.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: coord/src/search_cluster.c [hunk 1]
```diff
@@ -407,7 +407,7 @@ MRCommandGenerator SearchCluster_MultiplexCommand(SearchCluster *c, MRCommand *c
       size_t oldlen = strlen(cmd->strs[mux->keyOffset]);
```
Comment (nafraf) [inline coord/src/search_cluster.c:407]: Now we can remove the `oldlen` variable

File: coord/src/search_cluster.c [hunk 2]
```diff
@@ -407,7 +407,7 @@ MRCommandGenerator SearchCluster_MultiplexCommand(SearchCluster *c, MRCommand *c
...
       size_t newlen = 0;
       const char *target = lookupAlias(cmd->strs[mux->keyOffset], &newlen);
-      if (oldlen != newlen) {
+      if (strcmp(cmd->strs[mux->keyOffset], target) != 0) {
         mux->keyAlias = rm_strndup(target, newlen);
```
Comment (oshadmi) [inline coord/src/search_cluster.c:411]: need to be set to null if not reaching here

File: tests/pytests/test.py [hunk 3]
```diff
@@ -2351,6 +2351,10 @@ def testAlias(env):
     env.expect('ft.aliasdel', 'myIndex', 'yourIndex').raiseError()
     env.expect('ft.aliasdel', 'non_existing_alias').raiseError()
 
+    # Test index alias with the same length as the original (issue 5945)
```
Comment (oshadmi) [inline tests/pytests/test.py:2354]: MOD, not issue

**Review by nafraf — APPROVED** [review]

**Review by oshadmi — APPROVED** [review]
👍🏼

**Pull request merged**